File: Python Codes/1_Divide_Two_Integers.py
def divide(dividend, divisor):
    if dividend == -2**31 and divisor == -1:
        return 2**31 - 1

    sign = -1 if (dividend > 0) != (divisor > 0) else 1

    abs_dividend = abs(dividend)
    abs_divisor = abs(divisor)

    quotient = 0

    while abs_dividend >= abs_divisor:
        temp_divisor = abs_divisor
        count = 1

        while abs_dividend >= (temp_divisor << 1):
            temp_divisor <<= 1
            count <<= 1

        abs_dividend -= temp_divisor
        quotient += count

    return sign * quotient

def main():
    dividend = int(input("Enter the dividend: "))
    divisor = int(input("Enter the divisor: "))

    result = divide(dividend, divisor)
    print("Result:", result)

if __name__ == "__main__":
    main()

File: CPP Codes/1_Divide_Two_Integers.cpp
#include<iostream>
#include<climits>
using namespace std;

class Solution{
    public:
    int divide (int dividend , int divisor)
    {
        if(dividend==INT_MIN &&divisor==-1)
        {
            return INT_MAX;
        }

        bool isNegative=(dividend>0)!=(divisor>0);
        long long absdividend = labs(dividend);
        long long absdivisor = labs(divisor);
        int answer=0;

        while(absdividend>absdivisor)
        {
            long long tempDivisor= absdivisor;
            int count=1;
            while(absdividend >= (tempDivisor<<1))
            {
                tempDivisor<<=1;
                count<<=1; 
        }
        absdividend -= tempDivisor;
        answer += count;
    }
    return isNegative? -answer : answer;
}
};

int main()
{
   Solution solution;

    int dividend, divisor;
    cout << "Enter the dividend: ";
    cin >> dividend;
    cout << "Enter the divisor: ";
    cin >> divisor;

    int result = solution.divide(dividend, divisor);
    cout << "Result: " << result << endl;

    return 0;
}
